### ISSUE TITLE
lizardfs makesnapshot is not keeping change dates of folders and files

### ISSUE THREAD

**Issue by r5r3** [lizardfs/lizardfs#569]
When a snapshot ist created with `lizardfs makesnapshot`, then the change date of all files is updated:
```
stat original/file
  File: 'original/file'
  Size: 35142     	Blocks: 69         IO Block: 65536  regular file
Device: 2eh/46d	Inode: 2046928     Links: 1
Access: (0644/-rw-r--r--)  Uid: (10000/  user)   Gid: (10000/group)
Context: system_u:object_r:fusefs_t:s0
Access: 2017-05-24 15:59:02.000000000 +0200
Modify: 2016-05-31 09:29:03.000000000 +0200
Change: 2017-05-24 15:59:02.000000000 +0200
 Birth: -
```

```
stat file/in/snapshot
  File: 'file/in/snapshot'
  Size: 35142     	Blocks: 69         IO Block: 65536  regular file
Device: 2eh/46d	Inode: 2046928     Links: 1
Access: (0644/-rw-r--r--)  Uid: (10000/  user)   Gid: (10000/group)
Context: system_u:object_r:fusefs_t:s0
Access: 2017-05-24 15:59:02.000000000 +0200
Modify: 2016-05-31 09:29:03.000000000 +0200
Change: 2017-06-24 00:20:25.000000000 +0200
 Birth: -
```

2017-06-24 00:20:25 is the time when the snapshot was taken. At this time, the file was not changed. The updated change dates are an issue for backup tools. We use the IBM TSM System for backups of snapshots. Unfortunately, it tries to update all actually unchanged files because of their new change date. This leads to unnecessary long run times of the backup.   

For folders it is even worse, their modification times are updated. I don't know if this is also relevant for the backup procedure, but it is at least confusing.

**Comment by psarna** [comment]
change time (ctime) indicates when metadata of the file was changed. Making a snapshot is basically creating a new file under different name without copying underlying data. It makes perfect sense to set ctime for this new file, because its metadata did change.

A backup system should use mtime instead of ctime when it needs to decide if file *contents* should be backed up...

**Comment by r5r3** [comment]
The backup system is not updating the content of the file, it is updating the metadata. Unfortunately, this takes some time for several million files (and causes unnecessary workload for the backup system, which we don't own).

I can not fully agree with the statement that the metadata was changed. It was copied, but it should be exactly the same. Also, snapshots for other file-systems like btrfs or zfs don't change the metadata of files included in the snapshot. Therefore, it seems reasonable that a backup system assumes that it needs to update stored metadata of files when the metadata of these files has changed. 

Anyway, I can imagine that there are reasons to update the ctime during the snapshot procedure (even if non comes to my mind know), but I would really appreciate a possibility to avoid that. The snapshot command could have a command line option to prevent updating the ctime attribute. This would increase compatibility with existing backup software (my guess is, that we are not the only IBM users). 

Another example where unchanged metadata would be desirable: some people may use rsync (or compareable tools) to make backups in a payed cloud storage, but some storage providers may charge a fee for iops.

**Comment by psarna** [comment]
Fair enough. A complete solution would take some time and I cannot guarantee when/if it goes to our roadmap, but if anyone is interested in applying a patch and recompiling OR implementing something like that as a pull request, here's where ctimes are actually updated:
http://cr.skytechnology.pl:8081/#/c/3012/

**Comment by r5r3** [comment]
Thank you very much for providing this patch!
